### ISSUE TITLE
transition from `feedback()` to `showFeedback()` and `hideFeedback()` functions 

### ISSUE THREAD

**Issue by merlinoa** [merlinoa/shinyFeedback#19]
It is much more simple and flexible to just use `if` `else` statements than it is to pass a `condition` argument to `feedback()`.  So, in order to use `if` `else` statements and remove the need for the `condition` argument to `feedback()`, we are adding `showFeedback()` and `hideFeedback()` functions to `shinyFeedback`.  They will work like the following pseudocode:

```
observeEvent(input$my_input, {
  if (input$my_input > 5) {
    showFeedback("my_input", ...)
  } else {
    hideFeedback("my_input")
  }
})
```

As you can see in the above code, the "condition" of whether or not to show the feedback is now handled in the normal R control flow rather than with a "condition" argument to `feedback()`.

I am planning to remove `feedback()` once `showFeedback()` and `hideFeedback()` are up and running.  Please comment on this issue if you have suggestions.

**Comment by hadley** [comment]
I'm now not 100% sure that this is true, because I think you'll often want to use `showFeedback()` with `req()` (in order to stop further reactives from erroring), e.g.:

```R
server <- function(input, output, session) {
  data <- reactive({
    req(input$dataset)
    
    exists <- exists(input$dataset, "package:datasets")
    shinyFeedback::feedbackDanger("dataset", !exists, "Unknown dataset")
    req(exists, cancelOutput = TRUE)

    get(input$dataset, "package:datasets")
  })
  
  output$data <- renderTable({
    head(data())
  })
}
```

I think that's a little more elegant than the equivalent that uses `if`:

```R
server <- function(input, output, session) {
  data <- reactive({
    req(input$dataset)
    
    exists <- exists(input$dataset, "package:datasets")
    if (!exists) {
      shinyFeedback::showFeedback("dataset", "Unknown dataset")
      req(FALSE, cancelOutput = TRUE)
    } else {
      shinyFeedback::hideFeedback("dataset")
    }

    get(input$dataset, "package:datasets")
  })
  
  output$data <- renderTable({
    head(data())
  })
}
```

**Comment by merlinoa** [comment]
Thanks @hadley!  Yes I agree that your top example is a much more natural fit with `shiny::req()`.

So, `shinyFeedback::feedback()` has been added back to shinyFeedback (I removed it from the GitHub version a couple days ago).  To be consistent with `shiny::req()`, `feedback()` now uses `shiny::isTruthy()` to check whether or not the feedback message should be displayed.

Also thanks very much for the example code!  I added a slightly adapted version of it to the shinyFeedback intro vignette. 

All 3 functions, feedback()`, showFeedback()`, and `hideFeedback()`, will be in the next release of shinyFeedback().
